src/main/java/com/microsoft/lst_bench/common/SessionExecutor.java

```diff
@@ -112,6 +111,18 @@ private Map<String, Object> updateRuntimeParameterValues(TaskExec task) {
     return values;
   }
 
+  private TaskExecutor getTaskExecutor(TaskExec task) {
+    if (task.getCustomTaskExecutor() == null) {
+      return new TaskExecutor(this.telemetryRegistry, this.experimentStartTime);
+    }
+    switch (task.getCustomTaskExecutor()) {
+      case "DependentTaskExecutor":
+        return new DependentTaskExecutor(this.telemetryRegistry, this.experimentStartTime);
+      default:
+        return new TaskExecutor(this.telemetryRegistry, this.experimentStartTime);
```

Comment:
I think landing in `default` would be unexpected. If that is the case, consider throwing an `IllegalArgumentException`.